src/one_time/core.clj

```diff
@@ -4,6 +4,15 @@
             [one-time.totp  :as totp]
             [one-time.hotp  :as hotp]))
 
+(defn ^:private ct=
+  "Constant-time comparison for numeric types, via bytewise string equality"
+  [x y]
+  (let [x' (.getBytes (str x))
+        y' (.getBytes (str y))]
+    (and
+     (= (count x') (count y'))
```

Comment:
I've seen this short-circuit across more implementations (in other libs/langs), so I understand that it's generally accepted. 

From my understanding, with this in place 
- an attacker can possibly discover the length of data being compared
- it's not quite constant time (because of an obvious short circuit)

So, I'd prefer the function name and description to reflect that (perhaps we can call it secure comparison instead). Hopefully, you'll agree.

Reply:
It's certainly true the attacker can discover the length in question, but for TOTP that's generally public. It's pretty difficult to generally make that constant time (unless you just artificially wait some amount of time that you expect all comparisons to be done by): e.g. if you just extend the shortest version with zeroes to do a bunch of extra fake compares, you're still leaking however long those compares last.

I'll rename the fn either way, just adding that for context.